src/@next/Avatar/AvatarStyle.tsx

```diff
@@ -0,0 +1,88 @@
+import styled from 'styled-components';
+import * as Breakpoints from '../utilities/breakpoints';
+import { Blue, Green, Neutral, Orange, Red } from '../utilities/colors';
+import {
+  AvatarBackgroundColorVariant,
+  AvatarProps,
+  AvatarSizeVariant,
+} from './Avatar';
+
+const avatarBackgroundColor: {
+  [variant in AvatarBackgroundColorVariant]: string;
+} = {
+  ['supplementary']: Neutral.B99,
+  ['warning']: Orange.S42,
+  ['danger']: Red.B100,
+  ['success']: Green.B89,
+  ['outstanding']: Blue.S08,
+};
+
+const avatarSize: {
+  [variant in AvatarSizeVariant[number]]: number;
+} = {
+  ['large']: 60,
+  ['medium']: 40,
+};
+
+const mobileAvatarSize: {
+  [variant in AvatarSizeVariant[number]]: number;
+} = {
+  ['large']: 40,
+  ['medium']: 32,
+};
```

Comment:
can we do something like this for both medium and large styles:
```
const mediumAvatarSizeStyle = `
  width: 40px;
  height: 40px;
  
  @media (max-width: ${Breakpoints.large}) {
    width: 32px;
    height: 32px;
  }
`;
```
and map the style with the size so we eliminate the other called functions as much as we can inside the styled component.

Reply:
i did some refactoring of it, you can take a look

also, i've made sure that the snapshot didn't get updated to maintain the integrity of the outcome